lib/elixir/lib/calendar/date.ex

```diff
@@ -748,12 +748,11 @@ defmodule Date do
   end
 
   defimpl Inspect do
```

Comment:
On revisiting `Calendar.Holocene` I realised that with this change to the `Inspect/2` implementation there is now no API to emit a formatted string for a calendar that identifies itself - if we want to keep the ability to roundtrip which I believe we do.

For example: the new `calendar.to_string(date)`  contract emits an un-annotated string. Therefore is `"2000-01-01"` in `Calendar.ISO`, or `Calendar.Holocene`, or some other calendar?  Previously `to_string/1` returned "2000-01-01 (HE)" which I updated to "2000-01-01 Calendar.Holocene". The contract was that `Calendar.ISO` dates had no annotation but other calendars did.  This won't now work because `inspect/2` will add a duplicate calendar annotation.

I don't know what the right answer to this is. I really like the cleanliness of how `inspect/2` now works. But we lose the ability to `to_string/1` a date/time/date_time and be able to use that to directly disambiguate a date in a calendar.

Reply:
> For example: the new calendar.to_string(date) contract emits an un-annotated string. Therefore is "2000-01-01" in Calendar.ISO, or Calendar.Holocene, or some other calendar? Previously to_string/1 returned "2000-01-01 (HE)" which I updated to "2000-01-01 Calendar.Holocene". The contract was that Calendar.ISO dates had no annotation but other calendars did.

I don't think we lost anything with the new contract. If it makes sense for Holocene to keep the ` (HE)` suffix, that's still totally cool.

If we do `~D[2019-01-01 (HE) Calendar.Holocene]` then it will result in `Calendar.Holocene.parse_date("2019-01-01 (HE)")` which can look like this:

```elixir
defmodule Calendar.Holocene do
  def parse_date(string) do
    string = String.trim_trailing(string, " (HE)")
    Calendar.ISO.parse_date(string) # forgetting about 5 digit years for a sec!
  end

  def date_to_string(year, month, day) do
    Calendar.ISO.date_to_string(year, month, day) <> " (HE)"
  end
end
```

Reply:
I would also wonder if Holocene should really emit an annotated string. For example, would people using a Japanese calendar to see dates as "xyz Japanese Calendar"? `to_string` is always about how a non-programmer would interpret it.

Reply:
Yup, exactly. I think it's totally up to calendar implementer to add annotations, but I think that'd be rather uncommon.

Reply:
José, yes, I agree with that having had a few coffees now.   And if localised formatting is required (as is a more likely scenario) then I have a lib for that.  Unannotated makes more sense.

Just noting that calendars aren't always about culture, often its about business and fiscal calendars which, whilst still Gregorian, are based upon different start dates. But the ambiguity in presentation should be resolved separately, not by annotating `Kernel.to_string/1` (ie I agree with. your comment).

Properly armed I'm now going to update and greatly simplify my implementations. Thanks to you and @wojtekmach for being patient through this.

Now if I can just work our a compelling proposal for weeks and quarters I think I'll move on from calendars as a happy person :-)